LeaderboardBackend.Test/Categories.cs

```diff
@@ -97,6 +101,209 @@ public static async Task CreateCategory_GetCategory_OK()
+        problemDetails!.Title.Should().Be("Leaderboard Not Found");
+    }
+
+    [Test]
+    public static async Task CreateCategory_NoConflictBecauseOldCatIsDeleted()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Category cat = new()
+        {
+            Name = "First",
+            Slug = "should-not-conflict",
+            LeaderboardId = _createdLeaderboard.Id,
+            SortDirection = SortDirection.Ascending,
+            Type = RunType.Score,
+            DeletedAt = _clock.GetCurrentInstant(),
+        };
+
+        context.Categories.Add(cat);
+        await context.SaveChangesAsync();
+        cat.Id.Should().NotBe(default);
+
+        CreateCategoryRequest request = new()
+        {
+            Name = "Shouldn't conflict",
+            Slug = "should-not-conflict",
+            Info = "",
+            SortDirection = SortDirection.Ascending,
+            Type = RunType.Time
+        };
+
+        await FluentActions.Awaiting(() => _apiClient.Post<CategoryViewModel>(
+            $"/leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = request,
+                Jwt = _jwt
+            }
+        )).Should().NotThrowAsync();
+    }
+
+    [Test]
+    public static async Task CreateCategory_Conflict()
+    {
+        CreateCategoryRequest request = new()
+        {
+            Name = "First",
+            Slug = "repeated-slug",
+            Info = "",
+            SortDirection = SortDirection.Ascending,
+            Type = RunType.Time
+        };
+
+        CategoryViewModel created = await _apiClient.Post<CategoryViewModel>(
+            $"/leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = request,
+                Jwt = _jwt
+            }
+        );
+
+        ExceptionAssertions<RequestFailureException> exAssert = await FluentActions.Awaiting(() => _apiClient.Post<CategoryViewModel>(
+            $"/leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = request,
+                Jwt = _jwt,
+            }
+        )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Conflict);
+
+        ProblemDetails? problemDetails = await exAssert.Which.Response.Content.ReadFromJsonAsync<ProblemDetails>(TestInitCommonFields.JsonSerializerOptions);
+        problemDetails.Should().NotBeNull();
+        JsonElement serialised = problemDetails!.Extensions["conflict"].As<JsonElement>();
+        JsonSerializer
+            .Deserialize<CategoryViewModel>(serialised, TestInitCommonFields.JsonSerializerOptions)
+            .Should().Be(created);
+    }
+
+    [TestCase(null, "bad-data")]
+    [TestCase("Bad Data", null)]
+    // TODO: Figure out how to test against sort direction and run type. Passing
```

Comment:
Just go with it for now. Have expected status as a param to the test and use 400 for the cases with invalid enum values. Use an anonymous type instead of a CreateCategoryRequest.